tracing-opentelemetry/src/layer.rs

```diff
@@ -896,6 +942,52 @@ where
                 } else {
                     builder.events = Some(vec![otel_event]);
                 }
+
+                // limit the maximum number of events we'll record per-span
+                match (self.max_events_per_span, &mut builder.events) {
+                    (Some(max_num_events), Some(ref mut events))
+                        if events.len() > max_num_events =>
+                    {
+                        // drain half of the existing events, using ceiling division
+                        // to handle the case where max_num_events is 0 or 1
+                        //
+                        // note: this should be more efficient than just continuously
+                        // popping the first element
```

Comment:
curious, whats the motivation here over say, just saving the most recent events?

Reply:
Oh I misread, dropping the oldest events is definitely the best option here!